Terminal.Gui/Views/SpinnerView.cs

```diff
@@ -1,44 +1,271 @@
+		Arc,
+		Circle,
+		SquareCorners,
+		CircleQuarters,
+		CircleHalves,
+		Squish,
+		Toggle,
+		Toggle2,
+		Toggle3,
+		Toggle4,
+		Toggle5,
+		Toggle6,
+		Toggle7,
+		Toggle8,
+		Toggle9,
+		Toggle10,
+		Toggle11,
+		Toggle12,
+		Toggle13,
+		Arrow,
+		Arrow2,
+		Arrow3,
+		BouncingBar,
+		BouncingBall,
+		Smiley,
+		Monkey,
+		Hearts,
+		Clock,
+		Earth,
+		Material,
+		Moon,
+		Runner,
+		Pong,
+		Shark,
+		Dqpb,
+		Weather,
+		Christmas,
+		Grenade,
+		Point,
+		Layer,
+		BetaWave,
+		FingerDance,
+		FistBump,
+		SoccerHeader,
+		MindBlown,
+		Speaker,
+		OrangePulse,
+		BluePulse,
+		OrangeBluePulse,
+		TimeTravelClock,
+		Aesthetic,
+		Aesthetic2,
+	}
+#pragma warning restore CS1591 // Missing XML comment for publicly visible type or member
 
 	/// <summary>
-	/// A 1x1 <see cref="View"/> based on <see cref="Label"/> which displays a spinning
-	/// line character.
+	/// A <see cref="View"/> based on <see cref="Label"/> which displays (by default) a
+	/// spinning line character.
 	/// </summary>
 	/// <remarks>
 	/// By default animation only occurs when you call <see cref="View.SetNeedsDisplay()"/>.
 	/// Use <see cref="AutoSpin"/> to make the automate calls to <see cref="View.SetNeedsDisplay()"/>.
 	/// </remarks>
 	public class SpinnerView : Label {
-		private Rune [] _runes = new Rune [] { '|', '/', '\u2500', '\\' };
+
+		private const SpinnerStyle DEFAULT_STYLE = SpinnerStyle.Line;
+		private const int DEFAULT_DELAY = 130;
+
+		private bool _hasSpecial = false;
+		private bool _bounceReverse = false;
+		private SpinnerStyle _style;
 		private int _currentIdx = 0;
 		private DateTime _lastRender = DateTime.MinValue;
 		private object _timeout;
 
+		/// <summary>
+		/// Gets or sets the frames used to animate the spinner.
+		/// </summary>
+		public string [] Frames { get; set; }
+
 		/// <summary>
 		/// Gets or sets the number of milliseconds to wait between characters
-		/// in the spin.  Defaults to 250.
+		/// in the spin.  Defaults to the SpinnerStyle's Interval value.
 		/// </summary>
 		/// <remarks>This is the maximum speed the spinner will rotate at.  You still need to
 		/// call <see cref="View.SetNeedsDisplay()"/> or <see cref="SpinnerView.AutoSpin"/> to
 		/// advance/start animation.</remarks>
-		public int SpinDelayInMilliseconds { get; set; } = 250;
+		public int SpinDelayInMilliseconds { get; set; } = DEFAULT_DELAY;
+
+		/// <summary>
+		/// Gets or sets whether spinner should go through frames in reverse order.
+		/// If SpinBounce is true, this sets the starting order.
+		/// </summary>
+		public bool SpinReverse { get; set; } = false;
+
+		/// <summary>
+		/// Gets or sets whether spinner should go back and forth through frames rather than
+		/// going to the end and starting again at the beginning.
+		/// </summary>
+		public bool SpinBounce { get; set; } = false;
+
+		/// <summary>
+		/// Gets the size in characters of the spinner.
+		/// </summary>
+		public int SpinnerWidth { get => GetSpinnerWidth (); }
```

Comment:
I may be missing something, but why is this needed? Can't we just use `View.Width`?

Reply:
Removed.